src/Groups/pcgroup.jl

```diff
@@ -365,3 +365,57 @@ function pc_group(c::GAP_Collector)
   end
 end
 
+"""
+    letters(g::PcGroupElem)
+
+Return the letters of `g` as a list of integers, each entry corresponding to
+a group generator.
+
+# Examples
+```jldoctest
+julia> c = collector(2, Int);
+
+julia> Oscar.set_relative_orders!(c, [2, 3])
+
+julia> Oscar.set_conjugate!(c, 2, 1, [2 => 2])
+
+julia> gg = pc_group(c)
+Pc group of order 6
+
+julia> letters(gg[1]^5*gg[2]^-4)
+3-element Vector{Int64}:
+ 1
+ 2
+ 2
+```
+"""
+function letters(g::PcGroupElem)
+  w = GAPWrap.UnderlyingElement(GapObj(g))
+  return Vector{Int}(GAPWrap.LetterRepAssocWord(w))
+end 
+
+function Oscar.syllables(g::Union{PcGroupElem, SubPcGroupElem})
+  l = GAPWrap.ExtRepOfObj(GapObj(g))
+  @assert iseven(length(l))
+  return Pair{Int, ZZRingElem}[l[i-1] => l[i] for i = 2:2:length(l)]
+end
+
+# Convert syllables in canonical form into exponent vector
+#Thomas
+function exponent_vector(sylls::Vector{Pair{Int64, ZZRingElem}}, n)
+  res = zeros(ZZRingElem, n)
+  for pair in sylls
+    @assert res[pair.first] == 0 #just to make sure 
+    res[pair.first] = pair.second
+  end
+  return res
+end
+
+# Convert syllables in canonical form into group element
```

Comment:
why does it only work for syllables in canonical form? I would at least expect the function to test this requirement.
And I think this requirement makes it impossible to construct certain elements in certain groups. Consider `< a, b | a^2 = 1, b^2 = 1, aba = bab >` (Weyl group of type A2). The element `aba` has no representation in "syllables in canonical form" (or maybe I misunderstood what canonical form means here)

Reply:
The point is that there are different useful ways to encode a group element that is given by some kind of word. For general elements in finitely presented groups, there is no normal form, and an element is represented by a sequence of letters or syllables.
For elements in polycyclic groups, there is a normal form which corresponds to an exponent vector, and many low level functions deal just with these exponent vectors. In situations where one needs a word-like description of an element, it may be necessary to switch from the exponent vector to a syllable description and back. Often it is not needed to do actual computations (such as normalizing the data) in such a case. Testing whether the data are admissible can make sense; in this case, there should be a `check = false` option that allows one to skip these tests.

Reply:
Thanks for the explanation. I might have mixed up FP and PcGroups when thinking about this.